src/graphnet/models/graphs/graphs.py

```diff
@@ -100,3 +108,203 @@ def __init__(
+        graph = add_full_rrwp(graph, walk_length=self.walk_length)
+
+        return graph
+
+
+class KNNGraphRWSE(GraphDefinition):
+    """KNN Graph with random walk structural encoding."""
+
+    def __init__(
+        self,
+        detector: Detector,
+        node_definition: Optional[NodeDefinition] = None,
+        edge_definition: Optional[EdgeDefinition] = None,
+        input_feature_names: Optional[List[str]] = None,
+        dtype: Optional[torch.dtype] = torch.float,
+        perturbation_dict: Optional[Dict[str, float]] = None,
+        seed: Optional[Union[int, Generator]] = None,
+        nb_nearest_neighbours: int = 8,
+        columns: List[int] = [0, 1, 2],
+        walk_length: int = 8,
+        **kwargs: Any,
+    ) -> None:
+        """Construct k-nn graph representation.
+
+        Args:
+            detector: Detector that represents your data.
+            node_definition: Definition of nodes in the graph.
+            edge_definition: Definition of edges in the graph.
+            input_feature_names: Name of input feature columns.
+            dtype: data type for node features.
+            perturbation_dict: Dictionary mapping a feature name to a standard
+                               deviation according to which the values for this
+                               feature should be randomly perturbed. Defaults
+                               to None.
+            seed: seed or Generator used to randomly sample perturbations.
+                Defaults to None.
+            nb_nearest_neighbours: Number of edges for each node.
+                Defaults to 8.
+            columns: node feature columns used for distance calculation.
+                Defaults to [0, 1, 2].
+            walk_length: number of steps for the random walk.
+                Defaults to 8.
+        """
+        # Base class constructor
+        super().__init__(
+            detector=detector,
+            node_definition=node_definition or NodesAsPulses(),
+            edge_definition=edge_definition
+            or KNNEdges(
+                nb_nearest_neighbours=nb_nearest_neighbours,
+                columns=columns,
+            ),
+            dtype=dtype,
+            input_feature_names=input_feature_names,
+            perturbation_dict=perturbation_dict,
+            seed=seed,
+            **kwargs,
+        )
+        self.walk_length = walk_length
+
+    def forward(  # type: ignore
+        self,
+        input_features: np.ndarray,
+        input_feature_names: List[str],
+        **kwargs,
+    ) -> Data:
+        """Forward pass."""
+        graph = super().forward(input_features, input_feature_names, **kwargs)
+        ksteps = torch.arange(1, self.walk_length)
+        graph.rwse = get_rw_landing_probs(
+            ksteps=ksteps, edge_index=graph.edge_index, edge_weight=None
+        )
+        return graph
+
+
+class KNNGraphNoPE(GraphDefinition):
```

Comment:
Considering the very high level of similarity between this and `KNNGraph`, perhaps it would be beneficial to introduce a new argument to `KNNGraph` that would toggle between the vanilla KNNEdges and your new KNNDistanceEdges. For example, one could invent the argument `distance_as_edge_features: bool = False` (defaults to `KNNEdges`) and use `KNNDistanceEdges` if `True`.